src/test/java/io/bf2/kafka/authorizer/CustomAclAuthorizerTest.java

```diff
@@ -236,7 +243,44 @@ void testCreateAclsDeniedForInvalidPrincipal(String title, String principal) thr
                     .map(AclCreateResult::exception)
                     .map(Optional::get)
                     .allMatch(e -> e instanceof ApiException
-                            && CustomAclAuthorizer.CREATE_ACL_INVALID_PRINCIPAL.equals(e.getMessage())));
+                            && error.equals(e.getMessage())));
+        }
+    }
+
+    @Test
+    void testCreateAclsAllowedWhenPrincipalGivesThemselvesACLRules() throws IOException {
+        final List<CompletableFuture<AclCreateResult>> creationResults = new ArrayList<>();
+        final AclCreateResult value = new AclCreateResult(null);
+        creationResults.add(CompletableFuture.completedFuture(value));
+        creationResults.add(CompletableFuture.completedFuture(value));
```

Comment:
Is it deliberate to add `value` twice to the `creationResults`

Reply:
Nope, it was a misunderstanding since the test created two bindings I thought they were pushed through to the delagate as a list and we wanted 2 results back out of the mock.